### PR TITLE
HITL: server-side helpers to measure client latency

### PR DESCRIPTION
**Pull request by eundersander**
## Motivation and Context

Add ClientMessageManager.set_server_keyframe_id and RemoteGuiInput.pop_recent_server_keyframe_id to help the server measure client latency.

This feature will require follow-up work in the Unity client. `stub_client.html` provides a reference implementation.

In the example code below, roughly speaking, this is the latency between something happening in a sim_step on the server and the client's reaction (as GuiInput) being available in a later sim_step.

Usage: consider a server AppState that maintains a frame counter. It uses `ClientMessageManager.set_server_keyframe_id` on every frame. The client receives and renders a frame with serverKeyframeID=k. It should include `recentServerKeyframeId=k` in the next clientState it sends to the server. (See stub_client.html for an example.). In the server AppState, it can check `RemoteGuiInput.pop_recent_server_keyframe_id()` and compare it to its current frame counter.

Example usage in an AppState:
```
def __init__(...):
    self._frame_counter = 0

def sim_update(...):
    ...
    if self._app_service.remote_gui_input:
        recent_server_keyframe_id = self._app_service.remote_gui_input.pop_recent_server_keyframe_id()
        if recent_server_keyframe_id:
            print(f"client frame latency: {self._frame_counter - recent_server_keyframe_id}")

    if self._app_service.client_message_manager:
        self._app_service.client_message_manager.set_server_keyframe_id(self._frame_counter)
    self._frame_counter += 1
```

## How Has This Been Tested

Local testing with rearrange and pick_throw_vr AppStates. With stub HTML client. Regression testing with Unity in-editor client (although it doesn't yet implement the required client-side functionality for this feature).

## Types of changes

[HITL]

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-hitl/habitat_hitl/_internal/networking/keyframe_utils.py [hunk 1]
```diff
@@ -95,18 +95,12 @@ def ensure_dict(keyframe, key):
...
-            "navmeshVertices",
-            "isAppReady",
-        ]:
-            if message_key in inc_message:
-                ensure_dict(consolidated_keyframe, "message")
-                consolidated_keyframe["message"][message_key] = inc_message[
-                    message_key
-                ]
+        # add/update all messages
+        for message_key in inc_message:
```
Comment (eundersander) [inline habitat-hitl/habitat_hitl/_internal/networking/keyframe_utils.py:99]: It seems like a good general rule that consolidating messages simply means updating to the latest message in the latest incremental keyframe. Let's beware a future situation where a client must absolutely receive an entire sequence of messages (not just the latest one).

File: habitat-hitl/habitat_hitl/scripts/stub_client.html [hunk 2]
```diff
@@ -48,6 +50,18 @@
                 document.getElementById('serverPort').disabled = true;
                 document.getElementById('sendButton').disabled = false;
                 ws.send('client ready!');  // Send "client ready!" to the server
+
+                // Start sending messages at 10 Hz
+                sendIntervalId = setInterval(function() {
```
Comment (eundersander) [inline habitat-hitl/habitat_hitl/scripts/stub_client.html:55]: As part of this PR, the stub HTML client now sends client states at 10 Hz.

**Review by eundersander — COMMENTED** [review]
Some comments.

File: habitat-hitl/habitat_hitl/core/remote_gui_input.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 from habitat_hitl.core.gui_input import GuiInput
 
 
+# todo: rename to RemoteClientState
```
Comment (0mdc) [inline habitat-hitl/habitat_hitl/core/remote_gui_input.py:15]: 👍

**Review by 0mdc — APPROVED** [review]
LGTM!

**Pull request merged**